pkg/database/realm.go

```diff
@@ -55,22 +74,77 @@ type Realm struct {
 	// Relations to items that belong to a realm.
 	Codes  []*VerificationCode `gorm:"PRELOAD:false; SAVE_ASSOCIATIONS:false; ASSOCIATION_AUTOUPDATE:false, ASSOCIATION_SAVE_REFERENCE:false"`
 	Tokens []*Token            `gorm:"PRELOAD:false; SAVE_ASSOCIATIONS:false; ASSOCIATION_AUTOUPDATE:false, ASSOCIATION_SAVE_REFERENCE:false"`
+
+	// Populated by AfterFind
+	codeDuration     time.Duration
```

Comment:
I think it'd be better to define a custom `time.Duration` type and define the sql package scanner:

```
type Duration time.Duration

func (d Duration) Value() (driver.Value, error) {
    return driver.Value(int64(d)), nil
}

func (d *Duration) Scan(raw interface{}) error {
    switch v := raw.(type) {
    case int64:
        *d = Duration(v)
    case nil:
        *d = Duration(0)
    default:
        return fmt.Errorf("unknown duration type: %T", raw)
    }
    return nil
}

// 
LongCodeDuration Duration
```

Gorm uses go/sql under the hood, so it should work.